go/pkg/apis/enricher/framework/java/jboss_eap_detector.go

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc.
+ ******************************************************************************/
+package enricher
+
+import (
+	"context"
+	"encoding/xml"
+	"strings"
+
+	"github.com/redhat-developer/alizer/go/pkg/apis/model"
+	utils "github.com/redhat-developer/alizer/go/pkg/utils"
+)
+
+type JBossEAPDetector struct{}
+
+type JBoss_Standalone_Xml struct {
+	SocketBindingGroup struct {
+		SocketBinding []struct {
+			Name string `xml:"name,attr"`
+			Port string `xml:"port,attr"`
+		} `xml:"socket-binding"`
+	} `xml:"socket-binding-group"`
+}
+
+func (o JBossEAPDetector) GetSupportedFrameworks() []string {
+	return []string{"JBoss EAP"}
+}
+
+func (o JBossEAPDetector) DoFrameworkDetection(language *model.Language, config string) {
+	if hasFwk, _ := hasFramework(config, "org.jboss.eap.plugins"); hasFwk {
```

Comment:
Could we detect based on the presence of the artifactId too?

Reply:
Ok I'll try to follow different approach for this detector. I think this addition is on the right direction as it will improve the recognition of frameworks.

Reply:
I've added checks for `artifactId` and `groupId` for these two frameworks